RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -199,12 +195,14 @@ public class PumpOpsSynchronous: NSObject {
   }
 
   internal func getHistoryEventsSinceDate(startDate: NSDate) throws -> ([PumpEvent], PumpModel) {
-    
-    if !defaultWake() {
+
+    do {
+      try wakeup()
+    } catch _ as PumpCommsError {
       try scanForPump()
     }
-    
-    guard let pumpModelStr = getPumpModel() else {
+
+    guard let pumpModelStr = try? getPumpModel() else {
```

Comment:
`try?` isn't a great pattern in all cases because it ignores what the underlying exception was. Rather than re-think any logic, my goal was to keep the logic exactly the same but adjust the implementation.

They say that every time you write a `!`, Apple deletes a Radar.